File: android/src/main/kotlin/com/valentingrigorean/arcgis_maps_flutter/mapping/symbology/GraphicController.kt
package com.valentingrigorean.arcgis_maps_flutter.mapping.symbology

import android.content.Context
import com.valentingrigorean.arcgis_maps_flutter.convert.geometry.toGeometryOrNull
import com.valentingrigorean.arcgis_maps_flutter.map.SymbolVisibilityFilterController

class GraphicController(private val context: Context, id: String) : BaseGraphicController() {

    init {
        graphic.attributes["graphicId"] = id
    }

    override var isSelected: Boolean
        get() = graphic.isSelected
        set(selected) {
            graphic.isSelected = selected
        }

    override fun interpretGraphicController(
        data: Map<*, *>,
        symbolVisibilityFilterController: SymbolVisibilityFilterController?
    ) {
        super.interpretGraphicController(data, symbolVisibilityFilterController)

        data["geometry"]?.toGeometryOrNull()?.let {
            geometry = it
        }

        data["symbol"]?.let {
            graphic.symbol = SymbolFactory.createSymbol(context, it)
        }

        data["isSelected"]?.let {
            isSelected = it as Boolean
        }
    }
}

File: ios/Classes/Mapping/Symbology/GraphicController.swift
//
// Created by Valentin Grigorean on 29.05.2024.
//

import Foundation
import ArcGIS

class GraphicMarkerController: BaseGraphicController {
    init(graphicId: String) {
        super.init(graphics: Graphic(geometry: nil, attributes: ["graphicId": graphicId], symbol: nil))
    }

    var symbol: Symbol? {
        get {
            graphics.symbol
        }
        set {
            graphics.symbol = newValue
        }
    }
    
    override var isSelected: Bool{
        get{
            graphics.isSelected
        }
        set{
            graphics.isSelected = newValue
        }
    }
}
